Sources/HAP/Endpoints/pairSetup().swift

```diff
@@ -38,36 +38,48 @@ func pairSetup(device: Device) -> Application {
         var body = Data()
         guard
             (try? request.readAllData(into: &body)) != nil,
-            let data: PairTagTLV8 = try? decode(body),
+            let data: [PairTagTLV8] = try? decode(body),
             let sequence = data[.state]?.first.flatMap({ PairSetupStep(rawValue: $0) })
         else {
             return .badRequest
         }
-        let response: PairTagTLV8?
+        let response: [PairTagTLV8]?
         do {
             switch sequence {
+            // M1: iOS Device -> Accessory -- `SRP Start Request'
             case .startRequest:
                 let session = createSession()
                 response = try controller.startRequest(data, session)
                 connection.context[SESSION_KEY] = session
+            // M3: iOS Device -> Accessory -- `SRP Verify Request'
             case .verifyRequest:
                 let session = try getSession(connection)
                 response = controller.verifyRequest(data, session)
+            // M5: iOS Device -> Accessory -- `Exchange Request'
             case .keyExchangeRequest:
                 let session = try getSession(connection)
                 response = try controller.keyExchangeRequest(data, session)
+            // Unknown state - return error and abort
             default:
-                response = nil
+                response = try controller.unknownRequest(data)
             }
         } catch {
             logger.warning(error)
             connection.context[SESSION_KEY] = nil
             response = nil
         }
         if let response = response {
-            return Response(status: .ok, data: encode(response), mimeType: "application/pairing+tlv8")
+            let status: Response.Status
+            if let errorData = response[.error] {
+                status = .badRequest
```

Comment:
When an invalid PIN is entered, iOS expects the HTTP status code to be 200 -- not 400 (bad request).

Reply:
OK I've changed it to return 200 OK if there is a TVL8 response to send, as it was before